Firebase/Auth/Source/Utilities/FIRAuthDefaultUIDelegate.m

```diff
@@ -70,8 +70,26 @@ - (void)dismissViewControllerAnimated:(BOOL)flag completion:(nullable void (^)(v
       applicationClass = cls;
     }
   }
-  UIApplication *application = [applicationClass sharedApplication];
-  UIViewController *topViewController = application.keyWindow.rootViewController;
+
+  UIViewController *topViewController;
+  if (@available(iOS 13.0, *)) {
+    UIApplication *application = [applicationClass sharedApplication];
+    NSSet<UIScene *> * connectedScenes = application.connectedScenes;
+    for (UIScene *scene in connectedScenes) {
+      if ([scene isKindOfClass:[UIWindowScene class]]) {
+        UIWindowScene *windowScene = (UIWindowScene *)scene;
+        for (UIWindow *window in windowScene.windows) {
+          if (window.isKeyWindow) {
+            topViewController = window.rootViewController;
+          }
+        }
+      }
+    }
+  } else {
+    UIApplication *application = [applicationClass sharedApplication];
+    topViewController = application.keyWindow.rootViewController;
+  }
```

Comment:
Nit: moving this code out to a separate method may improve readability of the code.

Reply:
I tried while it seems the code blocks depend on each other and hard to be separated.